core/src/test/scala/unit/kafka/tools/StorageToolTest.scala

```diff
@@ -488,4 +488,26 @@ Found problem:
       assertEquals(1, exitStatus)
     }
   }
+
+  @Test
+  def testFormatMultiEmptyDirectory(): Unit = {
```

Comment:
There is already a similar test case `testFormatSucceedsIfAllDirectoriesAreAvailable`, so could you please add new check to `testFormatSucceedsIfAllDirectoriesAreAvailable`? for example, we can split the `stream#toString` by new line literal and then check the duplicate contents.

Reply:
> There is already a similar test case `testFormatSucceedsIfAllDirectoriesAreAvailable`, so could you please add new check to `testFormatSucceedsIfAllDirectoriesAreAvailable`? for example, we can split the `stream#toString` by new line literal and then check the duplicate contents.

I think the print order should be consistent with the configuration order,it's just as important as format once.
Additional,printing is done through callbacks,just change `HashMap` to `linkedHashMap` seems impossible.
We will inverted order,  so I think we should  change related `HashMap` and `HashSet` to `LinkedHashMap` and `LinkedHashSet`  and create new Copier in each loop. 
About test cases,I will do it

Reply:
If we want to keep the order consistency, the `logDirs`, `emptyLogDirs`, and others should be changed to `LinkedHashSet`/`LinkedHashMap`. Also, the following callers have to use similar collection to keep the order. It seems to me guaranteeing the "output order" is good but the fix will introduce many changes to `StorageTool`.

Can we focus on fix the "repeated format" in this PR? Also, we can file a ticket to get more discussion/consensus about "print consistent order"

Reply:
you are right, I should focus on this PR. I will update my code.